ui/src/tool-panel/switch/dialog/switch-draft-oid-field.tsx

```diff
@@ -81,82 +116,110 @@ export const SwitchDraftOidField: React.FC<SwitchDraftOidFieldProps> = ({
+        </span>
+    );
+
+    const helpComponent = spinnerIfLoading ?? oidOkIndicator;
+
     return existingSwitchOid !== undefined ||
         existingSwitchOidLoaderStatus !== LoaderStatus.Ready ? (
         <React.Fragment />
     ) : (
-        <FieldLayout
-            label={`${t('switch-dialog.switch-draft-oid')} *`}
-            errors={combinedErrors}
-            value={
-                draftOidFieldOpen ? (
+        <React.Fragment>
+            <FieldLayout
+                label={`${t('switch-dialog.switch-draft-oid')} *`}
+                errors={combinedErrors}
+                warnings={oidPresenceWarnings}
+                help={helpComponent}
+                value={
                     <React.Fragment>
-                        <TextField
-                            qa-id="switch-draft-oid"
-                            value={draftOid}
-                            onChange={(e) => setDraftOid(e.target.value)}
-                            hasError={isVisited && combinedErrors.length > 0}
-                            onBlur={visitField}
-                            wide
-                        />
-                        {loadingOidPresence && <Spinner inputField />}
-                        {!loadingOidPresence &&
-                            oidPresence &&
-                            oidIsLocallyValid &&
-                            existingInGeoviite && (
+                        <div className={styles['switch-edit-dialog__oid-input-grid']}>
+                            <React.Fragment>
+                                <TextField
+                                    qa-id="switch-draft-oid"
+                                    value={editingOid ? draftOid : ''}
+                                    disabled={!editingOid}
+                                    ref={oidFieldRef}
+                                    placeholder={
+                                        !editingOid ? t('switch-dialog.switch-draft-oid-unset') : ''
+                                    }
+                                    onChange={(e) => setDraftOid(e.target.value)}
+                                    hasError={isVisited && combinedErrors.length > 0}
+                                    onBlur={visitField}
+                                    wide
+                                />
+                            </React.Fragment>
+                            <div>
                                 <Link
-                                    className={styles['switch-edit-dialog__alert']}
-                                    onClick={() => {
-                                        setDraftOid('');
-                                        setDraftOidFieldOpen(false);
-                                        onEdit(existingInGeoviite.id);
-                                    }}>
-                                    {moveToEditLinkText(t, existingInGeoviite)}
+                                    style={{ float: 'right' }}
```

Comment:
Note to self: Tämä classiksi (ja tehdyksi ylipäätään jotenkin muuten kuin floatilla)